src/util/detect-bugs/adopted-style-sheet.js

```diff
@@ -0,0 +1,47 @@
+import gentleRegisterProperty from "../gentle-register-property.js";
+
+/**
+ * Detect if the browser is affected by the Safari adopted style sheet bug:
+ * removing a CSS custom property from an adopted stylesheet in a ShadowRoot
+ * does not update the computed value as expected and doesn't trigger a transition.
+ * @returns {{value: boolean, valuePending: boolean}}
+ */
+export default {
+	value: undefined,
+	get valuePending () {
+		if (this.value !== undefined) {
```

Comment:
It seems nothing in here is async? So this should probably be a self-overwriting getter on `value`, with `valuePending` just resolving to that. We can then tweak `detect-bugs` to synchronously resolve if `value` is available (if it doesn't do that already)

Reply:
> It seems nothing in here is async?

Yup, it's sync. I'll try to refactor it using your proposal. Thanks!